code/system/application/perception/src/detectlane/detectlane.cpp

```diff
@@ -70,266 +101,359 @@ void DetectLane::nextContainer(odcore::data::Container &c)
-  // OLA CHANGE HERE IF NEEDED
-  int T1 = 180;
-  cv::inRange(src, cv::Scalar(T1, T1, T1), cv::Scalar(255, 255, 255), image);
-  medianBlur(image, image, 3);
+Eigen::Matrix3d DetectLane::readreadThreeByThreeMatrix(std::string fileName)
+{
+  std::ifstream file(fileName);
 
-    
-  //-----------------------------
-  // Local line search
-  //-----------------------------
-  ScanImage(&image,&lines,&recoveredPoints,nPoints,MINROW,MAXROW);
-  
-  
-  //-----------------------------
-  // Extract lines from points
-  //-----------------------------
-  ExtractLines(&recoveredPoints,&K,&M,nRegions,nPoints,&pointsPerRegion);
+  Eigen::Matrix3d transformationMatrix;
 
-  //-----------------------------
-  // Make decision based on lines
-  //-----------------------------
-  laneLocation2 = Eigen::VectorXd::Zero(nRegions, 1);
-  SelectLanesV2(pointsPerRegion, laneLocation2);
-  
-  if (fabs(pointsPerRegion.sum()) > 0.0001){
-    SelectLaneOrientation(regionIndex,laneLocation2,(int)recoveredPoints.cols());
-    p1 = regionIndex(0,0), p2 = regionIndex(1,0);
-    
-    
-    
-    //-----------------------------
-    //Add momentum & update previous lines
-    //-----------------------------
-    
-    // AddMomentum(K,kPrev,M,mPrev,alpha,regionIndex);
-    // kPrev = K;
-    // mPrev = M;
-    // prevP1 = p1;
-    // prevP2 = p2;
-    
-    
-    //-----------------------------
-    // Draw boarders/lines - VISUALIZATION
-    //-----------------------------
-    
-    DrawBorders(&src,MINROW,MAXROW,K(p1,0),K(p2,0),M(p1,0),M(p2,0));
- //   DrawTracks(&src, &K, &M,MINROW,MAXROW,Scalar(0,0,255));
-   DrawTracks(&src, &k,&m,MINROW,Scalar(255,255,255));
-    
 
-    //-----------------------------
-    // Calculate lane offset
-    //-----------------------------
-    laneOffset = GetLateralPosition(K(p1,0),M(p1,0),
-                                    K(p2,0),M(p2,0),
-                                    lines.col(0)(midRegion),lines.col(1)(midRegion),
-                                    (MAXROW));
-    //-----------------------------
-    // Approximate heading angle
-    //-----------------------------
-    double d1 = GetLateralPosition(K(p1,0),M(p1,0),
-                                K(p2,0),M(p2,0),
-                              lines.col(0)(midRegion),lines.col(1)(midRegion),
-                                (MAXROW));
-    double d2 = GetLateralPosition(K(p1,0),M(p1,0),
-                                K(p2,0),M(p2,0),
-                                lines.col(0)(midRegion),lines.col(1)(midRegion),
-                                (MINROW));
-    
-    double theta = atan((d2-d1) / (double)15);
-    
-    // DEBUG PRINT
-    //std::cout<<"Heading angle "<<theta<<std::endl;
-    //std::cout<<"Offset "<<laneOffset<<std::endl;
-		
-    
-    // Send the message
-    opendlv::perception::LanePosition lanePosition(laneOffset,theta);
-    odcore::data::Container msg(lanePosition);  
-    getConference().send(msg);
-		
+  if (file.is_open())
+  {
```

Comment:
Brackets should be on the same line as the statement (exception after methods/functions, but they are not statements really)